services/auth.js

```diff
@@ -82,31 +87,41 @@ const authService = {
     return deleteInfo
   },
 
-  activate: async (activationLink) => {
-    const user = await User.findOne({ activationLink })
-    if (!user) {
-      throw createError(400, BAD_ACTIVATION_LINK)
+  confirmEmail: async (confirmToken) => {
+    if (!confirmToken) {
+      throw createUnauthorizedError()
+    }
+
+    const tokenData = tokenService.validateConfirmToken(confirmToken)
+    const tokenFromDB = await tokenService.findToken(confirmToken, CONFIRM_TOKEN)
+
+    if (!tokenFromDB || !tokenData) {
+      throw createError(401, BAD_CONFIRM_TOKEN)
     }
-    if (user.isActivated) {
-      throw createError(400, ALREADY_ACTIVATED)
+
+    const { id: userId } = tokenData
+    const user = await userService.getUser({ _id: userId })
+
+    if (user.isEmailConfirmed) {
+      throw createError(400, EMAIL_ALREADY_CONFIRMED)
     }
-    user.isActivated = true
-    await user.save()
+
+    await User.updateOne({ _id: userId }, { $set: { isEmailConfirmed: true } }).exec()
   },
 
   refresh: async (refreshToken) => {
     if (!refreshToken) {
       throw createUnauthorizedError()
     }
 
-    const userData = tokenService.validateRefreshToken(refreshToken)
-    const tokenFromDb = tokenService.findToken(refreshToken, REFRESH_TOKEN)
+    const tokenData = tokenService.validateRefreshToken(refreshToken)
+    const tokenFromDB = tokenService.findToken(refreshToken, REFRESH_TOKEN)
 
-    if (!userData || !tokenFromDb) {
+    if (!tokenData || !tokenFromDB) {
       throw createUnauthorizedError()
     }
 
-    const user = await User.findById(userData.id).populate('role').exec()
+    const user = await User.findById(tokenData.id).populate('role').exec()
```

Comment:
use userService.getUser()

Reply:
it will be fixed within another task, issue will be created shortly
thank you!